cpg_utils/hail_batch.py

```diff
@@ -70,20 +62,17 @@ def copy_common_env(job: hb.batch.job.Job) -> None:
 def remote_tmpdir(hail_bucket: Optional[str] = None) -> str:
     """Returns the remote_tmpdir to use for Hail initialization.
 
-    If `hail_bucket` is not specified explicitly, requires the HAIL_BUCKET environment variable to be set."""
-
-    if not hail_bucket:
-        hail_bucket = os.getenv('HAIL_BUCKET')
-        assert hail_bucket
-    return f'gs://{hail_bucket}/batch-tmp'
+    If `hail_bucket` is not specified explicitly, requires the `hail/bucket` config variable to be set.
+    """
+    return f'gs://{hail_bucket or _config["hail"]["bucket"]}/batch-tmp'
```

Comment:
I'm not sure that this is the best place for this, but some checking prior to reaching for the values could be useful?

```suggestion
    bucket = hail_bucket or _config.get('hail', {}).get('bucket')
    assert bucket, f'hail_bucket name was not set by argument or configuration'
    return f'gs://{bucket}/batch-tmp'
```

Reply:
Thanks, that's indeed much better!

Reply:
maybe an access method somewhere...

```
def _value_from_config(*args):
    assert args, '_value_from_config requires index arguments'
    _lookup = deepcopy(conf)
    for key in args:
        if key not in _lookup:
            raise Exception(f'key {key} was not present in the config dictionary. Requested index: {"::".join(args)}')
        _lookup = _lookup.get(key)

    return _lookup

conf = {'foo': {'foo2':'bar'}}
print(_value_from_config('foo', 'foo2')). # bar
print(_value_from_config('foo', 'bar'))
# Exception: key bar was not present in the config dictionary. Requested index: foo::bar
```

Reply:
While I like that idea, wouldn't a more generic solution be to use something like https://pypi.org/project/traceback-with-variables/ for the whole program?